### PR TITLE
SPOI-4384: atomic saving of checkpoint

### PR DESCRIPTION
**Pull request by chandnisingh**

### CODE REVIEW

**Comment by tweise** [comment]
Is the tmp file excluded when listing the checkpoints?

**Comment by chandnisingh** [comment]
I have added the code to ignore tmp files in case they exist while listing checkpoints.
tmp files are also being deleted when checkpoint is not saved successfully.

File: library/src/main/java/com/datatorrent/lib/util/FSStorageAgent.java [hunk 1]
```diff
@@ -37,6 +39,7 @@
  */
 public class FSStorageAgent implements StorageAgent, Serializable
 {
+  private static final String TMP_FILE = ".tmp";
```
Comment (243826) [inline library/src/main/java/com/datatorrent/lib/util/FSStorageAgent.java]: can you use the same extension as we have noticed while using the hadoop copyfromlocal command?

**Comment by tweise** [comment]
There is no need to delete the checkpoint when it could not be saved. This needs to work on hard stop. The replacement container needs to be able to override the leftover file and not see partially written checkpoints.

**Comment by chandnisingh** [comment]
Incorporated both the comments

**Pull request merged**